enumerable.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each(arr)
+    x = 0
+    while x < arr.length
+      yield(arr[x])
+      x += 1
+    end
+  end
+
+  def my_each_with_index(arr)
+    i = 0
+    while i < arr.length
+      yield(arr[i], i)
+      i += 1
+    end
+  end
+
+  def my_select(arr)
+    y = 0
+    new_arr = []
+    while y < arr.length
+      new_arr.push(arr[y]) if yield(arr[y])
+      y += 1
+    end
+    new_arr
+  end
```

Comment:
- [ ]  Enumerables `my_select` calls the given block once for each element in self

`[1,2,3,4,5].my_select{ |num| puts num }` gives `error`

- [ ]  Enumerables `my_select` returns an Enumerator if no block is given

`[1,2,3,4,5].my_select` should return `#<Enumerator: [1, 2, 3, 4, 5]:my_select>`